backend/pkg/config/tls.go

```diff
@@ -0,0 +1,61 @@
+package config
+
+import (
+	"crypto/tls"
+	"flag"
+
+	"github.com/twmb/tlscfg"
+)
+
+// TLS represents generic TLS configuration options.
+type TLS struct {
+	Enabled               bool   `yaml:"enabled"`
+	CaFilepath            string `yaml:"caFilepath"`
+	CertFilepath          string `yaml:"certFilepath"`
+	KeyFilepath           string `yaml:"keyFilepath"`
+	InsecureSkipTLSVerify bool   `yaml:"insecureSkipTlsVerify"`
+}
+
+// SetDefaults for the TLS config.
+func (c *TLS) SetDefaults() {
+	c.Enabled = false
+}
+
+// Validate the TLS config.
+func (*TLS) Validate() error {
+	return nil
```

Comment:
Wondering wehther we should call TLSConfig() here to check early whether the tls settings may be valid. Classic issues may be:

- Wrong path to certs
- Console user has no read permissions on the filesystem
- Certs are not in expected PEM format
- Wrong CA for given key pair

I'm unsure whether defaults are set before or after our call to `Validate()`, this may be important. WDYT?

Reply:
If we think calling TLSConfig() is problematic, we can still check if os.ReadFile() works if path is != "" for any of these.

Reply:
Was thinking of doing that. `RedpandaAdminAPITLS` that this replaces did in fact to that.
Added so Validate performs `TLSConfig()`